ocean_provider/utils/datatoken.py

```diff
@@ -65,21 +65,29 @@ def verify_order_tx(
 
     # check provider fees
     datatoken_contract = get_datatoken_contract(web3, datatoken_address)
+    provider_fee_order_log = None
     provider_fee_event_logs = datatoken_contract.events.ProviderFee().processReceipt(
         tx_receipt, errors=DISCARD
     )
+    # search in all provider_fee events until we have a match. if not, we don't have a valid event
+    # also, make sure that somebody is not spoofing provider fee event from another datatoken
+    for provider_fees_logs in provider_fee_event_logs:
+        try:
+            provider_data = json.loads(provider_fees_logs.args.providerData)
+            if (
+                provider_data["dt"].lower() == datatoken_address.lower()
```

Comment:
also, can you please check this key exists? It won't exist on older requests

Reply:
see answer above